cocos/core/game.ts

```diff
@@ -365,11 +365,14 @@ export class Game extends EventTarget {
     }
 
     /**
-     * @en Run the game frame by frame.
-     * @zh 执行一帧游戏循环。
+     * @en
+     * Run once game loop.
+     * @zh
+     * 执行一帧游戏循环。
+     * @param deltaTime 此次循环与上次循环间隔的时间，单位为毫秒。默认为自然流逝的时间。
      */
-    public step () {
-        legacyCC.director.mainLoop();
+    public step (deltaTime?: number) {
+        legacyCC.director.tick(deltaTime);
```

Comment:
We should also change `mainLoop` usage to `tick` in `runMainLoop` and calculate delta time in game directly.